Analysis/Tasks/PWGDQ/v0selector.cxx

```diff
@@ -42,19 +44,275 @@ using namespace o2::framework;
 using namespace o2::framework::expressions;
 using std::array;
 
-//using FullTracksExt = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov, aod::TracksExtended, aod::TrackSelection, aod::pidTPC, aod::pidTOF, aod::pidTOFbeta>;
-using FullTracksExt = soa::Join<aod::Tracks, aod::TracksExtra, aod::pidTOFbeta>;
+//using FullTracksExt = soa::Join<aod::Tracks, aod::TracksExtended,
+using FullTracksExt = soa::Join<aod::FullTracks, aod::TracksExtra, aod::TracksExtended,
+                                aod::pidTPCFullEl, aod::pidTPCFullMu, aod::pidTPCFullPi,
+                                aod::pidTPCFullKa, aod::pidTPCFullPr,
+                                aod::pidTOFFullEl, aod::pidTOFFullMu, aod::pidTOFFullPi,
+                                aod::pidTOFFullKa, aod::pidTOFFullPr, aod::pidTOFbeta>;
+
+//using FullTracksExt = soa::Join<aod::FullTracks, aod::TracksExtended>;
+
+namespace o2::aod
+{
+
+namespace reducedv0
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision); //!
+DECLARE_SOA_COLUMN(v0Pt, pt, float);            //!
+DECLARE_SOA_COLUMN(v0Eta, eta, float);          //!
+DECLARE_SOA_COLUMN(v0Phi, phi, float);          //!
+DECLARE_SOA_COLUMN(v0Mass, mass, float);        //!
+} //namespace reducedv0
+
+// basic track information
+DECLARE_SOA_TABLE(ReducedV0s, "AOD", "REDUCEDV0", //!
+                  o2::soa::Index<>, reducedv0::CollisionId, reducedv0::v0Pt, reducedv0::v0Eta, reducedv0::v0Phi, reducedv0::v0Mass);
+
+// iterators
+using ReducedV0 = ReducedV0s::iterator;
+} // namespace o2::aod
 
 struct v0selector {
 
-  void init(o2::framework::InitContext&)
+  //void init(o2::framework::InitContext&)
+  //{
+  //}
+  Produces<aod::ReducedV0s> v0Gamma;
+
+  enum { // Reconstructed V0
+    kUndef = -1,
+    kGamma = 0,
+    kK0S = 1,
+    kLambda = 2,
+    kAntiLambda = 3
+  };
+
+  float alphav0(const array<float, 3>& ppos, const array<float, 3>& pneg)
+  {
+    std::array<float, 3> pv0 = {ppos[0] + pneg[0], ppos[1] + pneg[1], ppos[2] + pneg[2]};
+    float momTot = RecoDecay::P(pv0);
+    float lQlNeg = RecoDecay::dotProd(pneg, pv0) / momTot;
+    float lQlPos = RecoDecay::dotProd(ppos, pv0) / momTot;
+    return (lQlPos - lQlNeg) / (lQlPos + lQlNeg); //longitudinal momentum asymmetry
+  }
+
+  float qtarmv0(const array<float, 3>& ppos, const array<float, 3>& pneg)
+  {
+    std::array<float, 3> pv0 = {ppos[0] + pneg[0], ppos[1] + pneg[1], ppos[2] + pneg[2]};
+    float momTot2 = RecoDecay::P2(pv0);
+    float dp = RecoDecay::dotProd(pneg, pv0);
+    return std::sqrt(RecoDecay::P2(pneg) - dp * dp / momTot2); //qtarm
+  }
+
+  float phivv0(const array<float, 3>& ppos, const array<float, 3>& pneg, const int cpos, const int cneg, const float bz)
+  {
+    // implementation taken from AliDielectronPair.cxx
+    float px1 = -9999., py1 = -9999., pz1 = -9999.;
+    float px2 = -9999., py2 = -9999., pz2 = -9999.;
```

Comment:
Are these variables needed, given that you have already the 3-momentum vectors in the arguments?